vignettes/tutorial2.Rmd

```diff
@@ -0,0 +1,447 @@
+
+For users that wish to customize the training process, a detailed explanation of the science in each step is included in this vignette.
+
+
+### One-step residual field generation
+
+You can generate any number of residual fields in a single command (shown for generating 3 fields):
+```
+residgrids <- generate.TP.resids(emulator, ngen = 3)
+```
+
+`residgrids` is a list of `ngen` entries, each containing a matrix of residuals [Ntime X 2Ngrid] for a single realization. Columns 1:Ngrid contain the temperature residuals for the realization. Columns (Ngrid+1):2Ngrid contain the precipitation residuals for the realization. 
+
+
+With the default inputs to `trainTP`, `tvarconvert_fcn = NULL` and `pvarconvert_fcn = log`, the residuals will be generated in `(temperature, log(precipitation))` space. 
+
+### One-step full field construction
+This can also be accomplished in one command:
+``` 
+fullgrids <- generate.TP.fullgrids(emulator, residgrids,
+                                   tgav = tgav,
+                                   tvarunconvert_fcn = NULL, 
+                                   pvarunconvert_fcn = exp,
+                                   reconstruction_function = pscl_apply)
+```
+
+Note that the inclusion of the argument `pvarunconvert_fcn = exp` means that the full grids are provided in `(temperature, precipitation)` space. The log(precipitation) transformation has been undone.
+
+
+## The science of each step
+
+### Setup (repeated)
+
+```{r paramsrepeat}
+## parameters for the code below.
+ngen <- 4             # number of fields to generate
+nplt <- ngen          # number of fields to plot
+exfld <- 20           # example field to plot from the time series
+set.seed(8675309)     # Set RNG seed so results will be reproducible
+```
+
+All of the data needed for this tutorial is installed with the package.
+```{r datadirrepeat}
+library('fldgen')
+datadir <- system.file('extdata', package='fldgen')
+```
+
+### Reading the ESM data
+
+The ESM temperature and precipitation fields should be in netCDF filea.  The function that reads it returns a "griddata" structure that contains the data, as well as some information about the grid, such as the latitude, longitude, and time coordinates and the vector of area weights needed to compute grid averages.
```

Comment:
Typo on the first line of the paragraph.  Also the raw versions of these files are easier to work with if you wrap your text to a reasonable number of columns (try clicking "view file" above for an illustration of what I mean).

```suggestion
The ESM temperature and precipitation fields should be in netCDF files.  The function that reads 
it returns a `griddata` structure that contains the data, as well as some information about the 
grid, such as the latitude, longitude, and time coordinates and the vector of area weights needed 
to compute grid averages.
```